flink-ai-flow/ai_flow/api/ops.py

```diff
@@ -0,0 +1,582 @@
+                  node_type=operation_type,
+                  **kwargs)
+    add_ai_node_to_graph(node, inputs=input_data)
+    outputs = node.outputs()
+    if 0 == output_num:
+        output: NoneChannel = outputs[0]
+        return output
+    else:
+        if 1 == len(outputs):
+            output: Channel = outputs[0]
+            return output
+        else:
+            output: List[Channel] = []
+            for i in outputs:
+                tmp: Channel = i
+                output.append(tmp)
+            return tuple(output)
+
+
+def action_on_event(job_name: Text,
+                    event_key: Text,
+                    event_value: Text,
+                    event_type: Text = UNDEFINED_EVENT_TYPE,
+                    condition: MetCondition = MetCondition.NECESSARY,
+                    action: TaskAction = TaskAction.START,
+                    life: EventLife = EventLife.ONCE,
+                    value_condition: MetValueCondition = MetValueCondition.EQUAL,
+                    namespace: Text = DEFAULT_NAMESPACE,
+                    sender: Text = None
+                    ):
+    """
+       Add user defined control logic.
+       :param job_name: The job name identify the job.
+       :param namespace: the project name
+       :param event_key: The key of the event.
+       :param event_value: The value of the event.
+       :param event_type: The Name of the event.
+       :param condition: The event condition. Sufficient or Necessary.
+       :param action: The action act on the src channel. Start or Restart.
+       :param life: The life of the event. Once or Repeated.
+       :param value_condition: The event value condition. Equal or Update. Equal means the src channel will start or
+                               restart only when in the condition that the notification service updates a value which
+                               equals to the event value under the specific event key, while update means src channel
+                               will start or restart when in the the condition that the notification service has a update
+                               operation on the event key which event value belongs to.
+       :param sender: The event sender identity. If sender is None, the sender will be dependency.
+       :return:None.
+       """
+    control_edge = ControlEdge(head=job_name,
+                               event_key=event_key,
+                               event_value=event_value,
+                               event_type=event_type,
+                               condition=condition,
+                               action=action,
+                               life=life,
+                               value_condition=value_condition,
+                               namespace=namespace,
+                               sender=sender
+                               )
+    default_graph().add_edge(job_name, control_edge)
+
+
+def action_on_model_version_event(job_name: Text,
```

Comment:
action_on_model_version_event, action_on_dataset_event are essentially built-in control dependencies based on some predefined events. We need to tell a complete story about those events, otherwise uses might not understand where are those events from.